strawberryfields/engine.py

```diff
@@ -469,12 +478,14 @@ def run(self, program, *, args=None, compile_options=None, **kwargs):
             program, args=args, compile_options=compile_options, **eng_run_options
         )
 
-        if isinstance(program, TDMProgram):
+        if not isinstance(program, collections.abc.Sequence) and program.type == "tdm":
```

Comment:
This was mainly because importing `TDMProgram` caused a circular import in other places and thus broke things. Here, actually, both should work equally well, but `program.type == "tdm"` is used elsewhere, so it was just to be a bit more consistent.